src/docgen/view/documentation.ts

```diff
@@ -145,78 +147,92 @@ export class Documentation {
     const assembliesDir = options?.assembliesDir ?? root;
 
     const { name } = JSON.parse(await fs.readFile(manifestPath, 'utf-8'));
-    return Documentation.forAssembly(name, assembliesDir, {
-      language: options?.language,
-      loose: options?.loose,
-    });
+    return Documentation.forAssembly(name, assembliesDir);
   }
 
   /**
    * Create a `Documentation` object for a specific assembly from a directory of assemblies.
    *
    * @param assemblyName - The assembly name.
    * @param assembliesDir - The directory containing the assemblies that comprise the type-system.
-   * @param options - Additional options.
    */
-  public static async forAssembly(assemblyName: string, assembliesDir: string, options?: DocumentationOptions): Promise<Documentation> {
-    return withTempDir(async (workdir: string) => {
-
-      // always better not to operate on an externally provided directory
-      await fs.copy(assembliesDir, workdir);
-
-      let transpile, language;
-
-      switch (options?.language ?? Language.TYPESCRIPT) {
-        case Language.PYTHON:
-          language = TargetLanguage.PYTHON;
-          transpile = new PythonTranspile();
-          break;
-        case Language.TYPESCRIPT:
-          transpile = new TypeScriptTranspile();
-          break;
-        case Language.JAVA:
-          language = TargetLanguage.JAVA;
-          transpile = new JavaTranspile();
-          break;
-        case Language.CSHARP:
-          language = TargetLanguage.CSHARP;
-          transpile = new CSharpTranspile();
-          break;
-        default:
-          throw new Error(`Unsupported language: ${options?.language}. Supported languages are ${Object.values(Language)}`);
-      }
-      const assembly = await createAssembly(assemblyName, workdir, options?.loose ?? true, language);
-      return new Documentation(assembly, transpile);
-    });
-
+  public static async forAssembly(assemblyName: string, assembliesDir: string): Promise<Documentation> {
+    return new Documentation(assemblyName, assembliesDir);
   }
 
+  private readonly cleanupDirectories: string[] = [];
+  private readonly assembliesCache: Map<string, reflect.Assembly> = new Map();
+
   private constructor(
-    public readonly assembly: reflect.Assembly,
-    private readonly transpile: Transpile,
-  ) {
-  }
+    private readonly assemblyName: string,
+    private readonly assembliesDir: string,
+  ) {}
 
   /**
    * Generate markdown.
    */
-  public render(options?: RenderOptions): Markdown {
-    const submodule = options?.submodule ? this.findSubmodule(this.assembly, options.submodule) : undefined;
+  public async render(options?: RenderOptions): Promise<Markdown> {
+
+    const language = options?.language ?? Language.TYPESCRIPT;
+    const loose = options?.loose ?? true;
```

Comment:
I feel like assigning a default value might make things a little less `?`-heavy... (Feel free to opt out of this).

```suggestion
  public async render(options: RenderOptions = {}): Promise<Markdown> {

    const language = options.language ?? Language.TYPESCRIPT;
    const loose = options.loose ?? true;
```

Reply:
I like it